agent/connect/ca/provider.go

```diff
@@ -7,6 +7,36 @@ import (
 
 //go:generate mockery -name Provider -inpkg
 
+// ProviderConfig encapsulates all the data Consul passes to `Configure` on a
+// new provider instance. The provider must treat this as read-only and make
+// copies of any map or slice if it might modify them internally.
+type ProviderConfig struct {
+	// ClusterID is the current Consul cluster ID.
+	ClusterID string
+
+	// Datacenter is the current Consul datacenter.
+	Datacenter string
+
+	// PrimaryDC is true when the CA instance is in the primary DC typically it
+	// may choose to act as a root in this case while secondaries are typically
+	// intermediate CAs. In some case the primary DC in Consul is an intermediate
+	// signed by some external CA along with that CA's public cert so the old name
+	// of `IsRoot` was misleading.
+	PrimaryDC bool
+
+	// RawConfig is the user configuration for the provider and is
+	// provider-specific to be interpretted as the provider wishes.
+	RawConfig map[string]interface{}
+
+	// State contains a the State the same provider last persisted. It is provided
```

Comment:
```suggestion
	// State contains a copy of the State the same provider last persisted. It is provided
```
I took a stab at patching up this sentence a bit.